resource-enrichment.tf

```diff
@@ -43,7 +34,7 @@ data "aws_iam_policy_document" "lightstep_assume_role_policy_with_xid" {
       variable = "sts:ExternalId"
 
       values = [
-        "${var.external_id}",
+        "${random_string.external_id.result}",
```

Comment:
question: Not new to this PR, but - I don't understand what the "condition" is doing here. Is it still necessary?

Reply:
The external id is pretty manual.  We're just slapping a condition on the policy that says the value of the variable `sts:ExternalId` that comes with the api call must equal the random value we are generating here.  ie - the external id check isn't some magic built in system feature.

Reply:
Ohhh, gotcha! I was trying to read this as some condition that applied to the terraform script itself. I get it now, thanks!